lib/bright_serializer/attribute.rb

```diff
@@ -14,12 +14,19 @@ def initialize(key, condition, entity, &block)
       @entity = entity ? Entity::Base.new(entity) : nil
     end
 
-    def serialize(object, params)
+    def serialize(class_instance, object, params)
       return unless object
 
       value =
         if @block
-          @block.arity.abs == 1 ? object.instance_eval(&@block) : object.instance_exec(object, params, &@block)
+          case @block.arity.abs
+          when 1
+            object.instance_eval(&@block)
+          when 2
+            object.instance_exec(object, params, &@block)
+          when 3
+            object.instance_exec(class_instance, object, params, &@block)
```

Comment:
Have you try to use `class_instance.instance_exec` I think going that way can simplify object passing.

We will probably not need to pass `class_instance` at all.

---

Also I if we need to pass a third argument I think I would place it in last position, it will be more consistent to always avec `object` first, `params` second, etc.